test/Streamly/Test/Data/Stream/Concurrent.hs

```diff
@@ -236,50 +242,78 @@ main = hspec
-               in cmp stream sortEq [0, 1, 2, 3]
-
-        prop1 "append right associated"
-            $ let stream =
-                    Stream.fromPure 0
-                        `Async.append2` (Stream.fromPure 1
-                        `Async.append2` (Stream.fromPure 2
-                        `Async.append2` Stream.fromPure 3))
-               in cmp stream sortEq [0, 1, 2, 3]
-
-        prop1 "append balanced"
-            $ let leaf x y = Stream.fromPure x `Async.append2` Stream.fromPure y
-                  leaf11 = leaf 0 1 `Async.append2` leaf 2 (3 :: Int)
-                  leaf12 = leaf 4 5 `Async.append2` leaf 6 7
-                  stream = leaf11 `Async.append2` leaf12
-               in cmp stream sortEq [0, 1, 2, 3, 4, 5, 6,7]
+        asyncSpec $
+            let appWith cfg = Async.combineWith cfg Stream.nil Stream.nil
+            in prop1 "mapM all configs append [] []"
+                    . cmp2 sortEq [] . appWith
+
+        asyncSpec $
+            let appWith cfg = Async.combineWith cfg Stream.nil (Stream.fromPure 1)
+            in prop1 "mapM all configs append [] [1]"
+                    . cmp2 sortEq [1] . appWith
+
+
+        asyncSpec $
+            let appWith cfg = Async.combineWith
+                        cfg Stream.nil (Stream.fromPure 1)
+            in prop1 "mapM all configs append [1] []"
+                    . cmp2 sortEq [1] . appWith
+
+        asyncSpec $
+            let appWith cfg = Async.combineWith
+                        cfg (Stream.fromPure 0) (Stream.fromPure 1)
+            in prop1 "mapM all configs append [0] [1]"
+                    . cmp2 sortEq [0, 1] . appWith
+
+        asyncSpec $
+            let appWith cfg =
+                    Async.combineWith cfg
+                        (Async.combineWith cfg (Stream.fromPure 0) Stream.nil)
+                        (Stream.fromPure 1)
+            in prop1 "mapM all configs append [0] [] [1]"
+                    . cmp2 sortEq [0, 1] . appWith
+
+        asyncSpec $
+            let appWith cfg =
+                    Async.combineWith cfg
+                        (Async.combineWith cfg
+                            (Async.combineWith cfg
+                                (Stream.fromPure 0) (Stream.fromPure 1))
+                            (Stream.fromPure 2))
+                        (Stream.fromPure 3)
+            in prop1 "mapM all configs append left associated"
+                    . cmp2 sortEq [0, 1, 2, 3] . appWith
+
+        asyncSpec $
+            let appWith cfg =
+                    Async.combineWith cfg
+                        (Stream.fromPure 0)
+                        (Async.combineWith cfg
+                            (Stream.fromPure 1)
+                            (Async.combineWith cfg
+                                (Stream.fromPure 2) (Stream.fromPure 3))
+                        )
+            in prop1 "mapM all configs append right associated"
+                    . cmp2 sortEq [0, 1, 2, 3] . appWith
```

Comment:
`combineWith` isn't encouraged to be used. I wonder why we are so extensively testing it.